src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/PuzzleTestController.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using DevAdventCalendarCompetition.Data;
+using DevAdventCalendarCompetition.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace DevAdventCalendarCompetition.Controllers
+{
+    public class PuzzleTestController : BaseTestController
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public PuzzleTestController(ApplicationDbContext context, UserManager<ApplicationUser> userManager) : base(context)
+        {
+            _userManager = userManager;
+        }
+
+        [HttpGet]
+        [CanStartTest(TestNumber = 7)]
+        public IActionResult Index()
+        {
+            var test = _context.Tests.FirstOrDefault(el => el.Number == 7);
+            return View(test);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> CheckGameStarted()
+        {
+            var user = await _userManager.Users.FirstOrDefaultAsync(u => u.Email == HttpContext.User.Identity.Name);
+
+            if (user != null)
+            {
+                var test = await _context.TestAnswers.FirstOrDefaultAsync(a => a.AnsweringTime == DateTime.MinValue && a.Answer == null && a.UserId == user.Id);
```

Comment:
did you mean _testAnswer_ ? It's better name

Reply:
Changed in layers refactoring commit.